src/yunohost/backup.py

```diff
@@ -1214,12 +1232,44 @@ def _restore_system(self):
 
         regen_conf()
 
+        # Check if we need to do the migration 0009 : setup group and permission
+        # Legacy code
+        result = ldap.search('ou=groups,dc=yunohost,dc=org',
+                             '(&(objectclass=groupOfNamesYnh)(cn=all_users))',
+                             ['cn'])
+        if not result:
+            from yunohost.tools import _get_migration_by_name
+            setup_group_permission = _get_migration_by_name("setup_group_permission")
+            # Update LDAP schema restart slapd
+            logger.info(m18n.n("migration_0011_update_LDAP_schema"))
+            regen_conf(names=['slapd'], force=True)
+            setup_group_permission.migrate_LDAP_db()
+
+        # Remove all permission for all app which sill in the LDAP
+        for per in ldap.search('ou=permission,dc=yunohost,dc=org',
+                               '(&(objectClass=permissionYnh)(!(cn=main.mail))(!(cn=main.metronome))(!(cn=main.sftp)))',
+                               ['cn']):
+            if not ldap.remove('cn=%s,ou=permission' % per['cn'][0]):
+                raise YunohostError('permission_deletion_failed', permission=permission, app=app)
```

Comment:
There are several issues in this file ... e.g. here `permission` and `app` are not defined in this context :/ 

And a few line below, `permission_name` and `app_name` aren't defined either :/

Reply:
Normally this should fix the issue:

```suggestion
                raise YunohostError('permission_deletion_failed', permission=per['cn'][0].split('.')[0], app=per['cn'][0].split('.')[1])
```

Reply:
Should be fixed by https://github.com/YunoHost/yunohost/pull/763

Reply:
I have to admit that this is not really the more readable code :x (but it might be in context)